app/models/admin.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+# Define the Admin model class.
+class Admin < ApplicationRecord
+  has_secure_password
+
+  EMAIL_REGEX = /\A[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}\Z/i.freeze
+  FORBIDDEN_USERNAMES = %w[fstuer jlove].freeze
+
+  validates :username, length: { within: 8..25 },
+                       uniqueness: true
```

Comment:
here you validate that the username is unique so you can use find or find_by in the contorller